### PR TITLE
Re-enable sha

### PR DESCRIPTION
**Pull request by Don-Ward**
The first commit puts in "the plumbing" for cnv_c_str on Windows.
The second commit enables the SHA plugin.

### CODE REVIEW

File: ipl/cfuncs/icall.h [hunk 1]
```diff
@@ -170,6 +174,7 @@
 #define Type(d) ((int)((d).dword & TypeMask))
 
 typedef long word;
+typedef unsigned long uword;
```
Comment (Jafaral) [inline ipl/cfuncs/icall.h:177]: The way we are handling this type (size) has the potential of breaking things in unexpected ways on some platforms. We probably should be consistent with what the  runtime expect:

https://github.com/uniconproject/unicon/blob/master/src/h/typedefs.h#L18

**Review by Jafaral — COMMENTED** [review]
Looks good to me, but we need to fix the word/uword otherwise I know that wont be working in Windows at least.

File: plugins/src/SHA/RFC6234/makefile [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #CFLAGS = -D ORIGINAL_SHA_CODE_NOT_THREAD_SAFE -D USE_32BIT_ONLY
+CFLAGS = -fPIC -O2
+CC = gcc
```
Comment (Jafaral) [inline plugins/src/SHA/RFC6234/makefile:3]: Why are we hard-coding these ? Why not including the pulgin src make file ?

**Review by Jafaral — CHANGES_REQUESTED** [review]
makefile should  renamed to Makefile to be consist with other make files in Unicon.

**Comment by Don-Ward** [comment]
Hi Jafar,

Inline ..


> In plugins/src/SHA/RFC6234/makefile <https://github.com/uniconproject/unicon/pull/260#discussion_r770713062>:
> > @@ -1,4 +1,6 @@
>  #CFLAGS = -D ORIGINAL_SHA_CODE_NOT_THREAD_SAFE -D USE_32BIT_ONLY
> +CFLAGS = -fPIC -O2
> +CC = gcc
> Why are we hard-coding these ? Why not including the pulgin src make file ?
> 
At present, the RFC6234 files do not compile with -std=c11. I have yet to investigate why not (I was more interested in getting the SHA plugin to build at all).

> In ipl/cfuncs/icall.h <https://github.com/uniconproject/unicon/pull/260#discussion_r770702843>:
> 
> > @@ -170,6 +174,7 @@
>  #define Type(d) ((int)((d).dword & TypeMask))
>  
>  typedef long word;
> +typedef unsigned long uword;
> The way we are handling this type (size) has the potential of breaking things in unexpected ways on some platforms. We probably should be consistent with what the runtime expect:
> 
> https://github.com/uniconproject/unicon/blob/master/src/h/typedefs.h#L18 <https://github.com/uniconproject/unicon/blob/master/src/h/typedefs.h#L18>This isn’t a new problem, it was already there with the existing definition of word in icall.h

I’m not quite sure what you want to do — I’m guessing that you want to reflect the definitions in typedefs.h in call.h (both versions). The question is why that wasn’t done in the first place with the definition of word in icall.h. Do we have access to IntBits?

See
ca14d877  08-Feb-2017 | Another round of plugins infrastructure [Jafar Al-Gharaibeh]
which is where the definition was introduced.

Don

**Pull request merged**